graphql/schemabuilder/pagination.go

```diff
@@ -303,6 +303,27 @@ func (funcCtx *funcContext) consumePaginatedArgs(sb *schemaBuilder, in []reflect
 
 }
 
+func (sb *schemaBuilder) getKeyFieldOnStruct(nodeType reflect.Type) (string, error) {
+
+	nodeObj := sb.objects[nodeType]
+	if nodeObj == nil && nodeType.Kind() == reflect.Ptr {
+		nodeObj = sb.objects[nodeType.Elem()]
+	}
+	nodeKey := reverseGraphql(nodeObj.key)
+	if nodeKey == "" {
+		return nodeKey, fmt.Errorf("a key field must be registered for paginated objects")
+	}
+	if nodeType.Kind() == reflect.Ptr {
+		nodeType = nodeType.Elem()
+	}
+	if _, ok := nodeType.FieldByName(nodeKey); !ok {
+		return nodeKey, fmt.Errorf("field doesn't exist on struct")
```

Comment:
this check seems redundant? reflect will fail before this

Reply:
is there a test verifying that behavior?

Reply:
can't be tested with the current reflect check but nice to leave it in